sentry_sdk/client.py

```diff
@@ -115,6 +116,12 @@ def _get_options(*args, **kwargs):
 
     if rv["instrumenter"] is None:
         rv["instrumenter"] = INSTRUMENTER.SENTRY
+    else:
+        warnings.warn(
+            "The `instrumenter` option will be removed in the next major version.",
+            DeprecationWarning,
+            stacklevel=3,
+        )
```

Comment:
This will also emit deprecation warnings for people using OTel now, right? The problem with that is that there is nothing actionable for the user here -- they can't do anything to get rid of the warning. So they'll be pestered by this message that basically says "you're using the SDK wrong" and can't do anything about it.

I'd either get rid of the warning completely, or give it a bit more context so that folks know what to expect ("...the option will be removed since OpenTelemetry tracing will become the default" or something like that).

Reply:
Yeah fair point, I am not really sure the best way to go about this. The downside of no warning is that we will drop this option without ever having had a depreciation warning for it – the downside of including the warning is that people might be confused about what action they need to take. 

Maybe we should just highlight in the docs that `instrumenter` will be removed in the next major? Perhaps, we can also plan to add a deprecation warning in the final 2.x minor/patch release

Reply:
I like your suggestions!

Reply:
Okay, this is what I will do: I will split the deprecation warning into a separate PR that we can keep around for when we add it in the final 2.x release. This PR will only contain the docstring updates to clarify that we will be dropping `instrumenter` in 3.x. Then, I will prepare a docs PR

Reply:
Deprecation PR is here: #3308